content/blog/peace-of-mind-with-cloud-secret-providers/index.md

```diff
@@ -0,0 +1,669 @@
+```python
+import pulumi
+from pulumi_aws import kms
+
+# Create a new KMS key
+key = kms.Key("stack-encryption-key",
+    deletion_window_in_days=10,
+    description="KMS key for encrypting Pulumi secret values"
+)
+
+// Create an alias to the key
+alias = kms.Alias("alias/stack-encryption-key",
+    target_key_id=key.key_id
+)
+
+# Export the arns
+pulumi.export('key_arn',  key.arn)
+```
+{{% /choosable %}}
+
+
+{{% choosable language go %}}
+```go
+package main
+
+import (
+	"github.com/pulumi/pulumi-aws/sdk/v2/go/aws/kms"
+	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+
+		// Create a new KMS key
+		key, err := kms.NewKey(ctx, "stack-encryption-key", &kms.KeyArgs{
+			Description: pulumi.String("KMS key for encrypting Pulumi secret values"),
+			DeletionWindowInDays: pulumi.Int(10),
+		})
+		if err != nil {
+			return err
+		}
+
+		// Create an alias to the key
+		alias, err := kms.NewAlias(ctx, "alias/stack-encryption-key", &kms.AliasArgs{
+			TargetKeyId: key.KeyId,
+		})
+		if err != nil {
+			return err
+		}
+
+		// Export the arns
+		ctx.Export("keyArn", key.Arn)
+		ctx.Export("keyAlias", alias.Arn)
+		return nil
+	})
+}
+```
+{{% /choosable %}}
+
+{{% choosable language csharp %}}
+```csharp
+using Pulumi;
+using Kms = Pulumi.Aws.Kms;
+
+class KeyStack : Stack
+{
+    public KeyStack()
+    {
+        // Create a new KMS key
+        var key = new Kms.Key("stack-encryption-key", new Kms.KeyArgs
+        {
+            DeletionWindowInDays = 10,
+            Description = "KMS key for encrypting Pulumi secret values"
+        });
+
+        // Create a new alias to the key
+        var alias = new Kms.Alias("alias/stack-encryption-key", new Kms.AliasArgs
+        {
+            TargetKeyId = key.KeyId
+        });
+
+        this.KeyArn = key.Arn;
+        this.AliasArn = alias.Arn;
+    }
+
+    [Output("keyArn")] public Output<string> KeyArn { get; set; }
+    [Output("aliasArn")] public Output<string> AliasArn { get; set; }
+}
+```
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+Creating the key is enough to allow us to start using it for our encryption provider. However, we need to consider who has access to the key before we start utilizing it with Pulumi.
+
+## Scoping Permission to our Key
+
+Generally, in AWS, you scope access to resources using IAM roles. However, for sensitive values like KMS keys, IAM roles alone aren't enough to provide the security you might need. As an example, if someone in your AWS account as an IAM role with the following policy attached:
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "Stmt1588215924595",
+      "Action": "kms:*",
+      "Effect": "Allow",
+      "Resource": "*"
+    }
+  ]
+}
+```
+
+That user gets access to every KMS key in your account, which would also mean they could decrypt any secret in your Pulumi stack.
+
+To rectify this, we need to attach a Key Policy to the key:
```

Comment:
Perhaps make it clear that we are updating our code from before - not deploying a whole separate thing?